src/pages/TokenDetails/index.tsx

```diff
@@ -137,22 +119,7 @@ export default function TokenDetails() {
         <>
           <TokenDetail address={tokenAddress} />
           <RightPanel>
-            <SwapWidget
-              defaultChainId={connectedChainId}
-              defaultInputTokenAddress={'NATIVE'}
-              defaultOutputTokenAddress={tokenAddress}
-              hideConnectionUI
-              jsonRpcUrlMap={RPC_URL_MAP}
-              locale={locale}
-              onTxSubmit={onTxSubmit}
-              onTxSuccess={onTxSuccess}
-              onTxFail={onTxFail}
-              provider={provider}
-              routerUrl={ROUTER_URL}
-              theme={widgetTheme}
-              // tokenList={[]}
-              width={WIDGET_WIDTH}
-            />
+            {token && <Widget defaultToken={token} />}
```

Comment:
is the conditional render necessary?

Reply:
Widget is typed to require `Currency`, not `Currency | undefined`, because the passed prop only defines the starting currency. If `defaultToken` started as `undefined` and then became defined (as happens when currencies load) then the widget would be initialized with `undefined` and not render the token when it loads. (This is intended, as the widget is spec'ed to be able to swap any token after initialization.) Because of this, the conditional render is necessary; because it's enforced by the type of Widget's props, I don't have a comment there.

Reply:
I'll add a comment to `WidgetProps` explaining this behavior.

Reply:
I've tested, and I believe I was wrong here. I took out the conditional render.